### PR TITLE
Add table turbot_policy_value closes #14

### LINKED ISSUE #14: Add table turbot_policy_value
**Use Case**
I would like to write a Turbot Workspace Health mod.  Determining the number of policy values in `tbd`, `invalid`, and `error`, and their age is very important to overall workspace health.  Please include a `metadata` and `workspace` columns in this table.

### PR DESCRIPTION
**Pull request by ParthaI**
# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  id,
  state,
  is_default,
  is_calculated,
  policy_type_id,
  policy_value_type_mod_uri
from
  turbot_policy_value
where
  policy_type_id = 190630193572596;
+-----------------+-------+------------+---------------+-----------------+---------------------------+
| id              | state | is_default | is_calculated | policy_type_id  | policy_value_type_mod_uri |
+-----------------+-------+------------+---------------+-----------------+---------------------------+
| 243392852690564 | ok    | true       | false         | 190630193572596 | tmod:@turbot/gcp-iam      |
| 243392852612714 | ok    | true       | false         | 190630193572596 | tmod:@turbot/gcp-iam      |
| 197439793364380 | ok    | true       | false         | 190630193572596 | tmod:@turbot/gcp-iam      |
+-----------------+-------+------------+---------------+-----------------+---------------------------+
```

```
> select
  id,
  state,
  is_default,
  is_calculated,
  resource_id,
  policy_value_type_mod_uri
from
  turbot_policy_value
where
  resource_id = 243392852451917;
+-----------------+-------+------------+---------------+-----------------+---------------------------+
| id              | state | is_default | is_calculated | resource_id     | policy_value_type_mod_uri |
+-----------------+-------+------------+---------------+-----------------+---------------------------+
| 243392852558428 | ok    | true       | false         | 243392852451917 | tmod:@turbot/gcp-iam      |
| 243392852610665 | ok    | true       | false         | 243392852451917 | tmod:@turbot/gcp-cisv1    |
| 243392852586083 | ok    | true       | false         | 243392852451917 | tmod:@turbot/gcp-iam      |
| 243392852612714 | ok    | true       | false         | 243392852451917 | tmod:@turbot/gcp-iam      |
| 243392852590180 | ok    | true       | false         | 243392852451917 | tmod:@turbot/gcp-iam      |
| 243392852597350 | ok    | true       | false         | 243392852451917 | tmod:@turbot/gcp-cisv1    |
| 243392852602472 | ok    | true       | false         | 243392852451917 | tmod:@turbot/gcp-cisv1    |
| 243392852576866 | ok    | true       | false         | 243392852451917 | tmod:@turbot/gcp-iam      |
+-----------------+-------+------------+---------------+-----------------+---------------------------+
```

```
> select
  id,
  state,
  is_default,
  is_calculated,
  resource_type_id,
  policy_value_type_mod_uri
from
  turbot_policy_value
where
  resource_type_id = 190630180698932;
+-----------------+-------+------------+---------------+------------------+---------------------------+
| id              | state | is_default | is_calculated | resource_type_id | policy_value_type_mod_uri |
+-----------------+-------+------------+---------------+------------------+---------------------------+
| 243392852686466 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-cisv1    |
| 243392852586083 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-iam      |
| 243392852690564 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-iam      |
| 243392852688515 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-cisv1    |
| 243392852576866 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-iam      |
| 243392852558428 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-iam      |
| 243392852610665 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-cisv1    |
| 243392852646514 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-iam      |
| 197439793404331 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-cisv1    |
| 243392852612714 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-iam      |
| 243392852674176 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-cisv1    |
| 197439793401256 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-cisv1    |
| 243392852665974 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-iam      |
| 243392852590180 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-iam      |
| 197439793397157 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-cisv1    |
| 243392852670072 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-iam      |
| 243392852602472 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-cisv1    |
| 197439793357209 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-iam      |
| 243392852660851 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-iam      |
| 243392852597350 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-cisv1    |
| 197439793364380 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-iam      |
| 197439793371551 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-iam      |
| 197439793393058 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-iam      |
| 197439793345941 | ok    | true       | false         | 190630180698932  | tmod:@turbot/gcp-iam      |
+-----------------+-------+------------+---------------+------------------+---------------------------+
```

</details>

### CODE REVIEW

File: docs/tables/turbot_policy_value.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Table: turbot_resource
```
Comment (cbruno10) [inline docs/tables/turbot_policy_value.md:1]:
```suggestion
# Table: turbot_policy_value
```

File: docs/tables/turbot_policy_value.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+Resources in Turbot represent cloud configuration items such as users,
+networks, servers, etc.
+
+It is recommended that queries to this table should include (usually in the `where` clause) at least one
```
Comment (cbruno10) [inline docs/tables/turbot_policy_value.md:6]: I don't think this is relevant to this table?

File: docs/tables/turbot_policy_value.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+of these columns: `id`, `resource_type_id`, `resource_type_uri` or `filter`.
+
+A Policy Value is the effective policy setting on an instance of a resource type. Every resource that is targeted by a given policy setting will have its own value for that policy, which will be the resultant calculated policy for the "winning" policy in the hierarchy.
```
Comment (cbruno10) [inline docs/tables/turbot_policy_value.md:9]:
```suggestion
A policy value is the effective policy setting on an instance of a resource type. Every resource that is targeted by a given policy setting will have its own value for that policy, which will be the resultant calculated policy for the "winning" policy in the hierarchy.
```

File: docs/tables/turbot_policy_value.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+Policy settings are inherited through the resource hierarchy, and values for a resource are calculated according to policy settings at or above it in the resource hierarchy. For example, a policy setting at the Turbot level will be inherited by all resources below.
+
+It is recommended that queries to this table should include (usually in the `where` clause) at least one
+of these columns: `state`,`policyTypeId`, `resource_type_id`, `resource_type_uri` or `filter`.
+
+## Examples
+
+### List policy values by policy type id
```
Comment (cbruno10) [inline docs/tables/turbot_policy_value.md:18]:
```suggestion
### List policy values by policy type ID
```

File: docs/tables/turbot_policy_value.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  is_calculated,
+  policy_type_id,
+  type_mod_uri
+from
+  turbot_policy_value
+where
+  policy_type_id = 123456789;
+```
+
+### List policy values by resource id
```
Comment (cbruno10) [inline docs/tables/turbot_policy_value.md:34]:
```suggestion
### List policy values by resource ID
```

File: docs/tables/turbot_policy_value.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  id,
+  state,
+  is_default,
+  is_calculated,
+  policy_type_id,
+  type_mod_uri
+from
+  turbot_policy_value
+where
+  policy_type_id = 123456789;
```
Comment (cbruno10) [inline docs/tables/turbot_policy_value.md:31]: Can you please use a value here that looks like a policy type ID? I understand it won't match to a real resource ID when running it, but please see other table docs as an example

File: docs/tables/turbot_policy_value.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+  id,
+  state,
+  is_default,
+  is_calculated,
+  resource_id,
+  type_mod_uri
+from
+  turbot_policy_value
+where
+  resource_id = 123456789;
```
Comment (cbruno10) [inline docs/tables/turbot_policy_value.md:47]: Can you please use a value here that looks like a resource type ID? I understand it won't match to a real resource ID when running it, but please see other table docs as an example

File: docs/tables/turbot_policy_value.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+```
+
+### List policy values by resource type id
```
Comment (cbruno10) [inline docs/tables/turbot_policy_value.md:50]: Do we need this example, and the one above? Is one more useful than the other?

File: docs/tables/turbot_policy_value.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+  state,
+  is_default,
+  is_calculated,
+  policy_type_id,
+  resource_id,
+  resource_type_id
+from
+  turbot_policy_value
+where
+  filter = 'policyTypeId:123456789';
```
Comment (cbruno10) [inline docs/tables/turbot_policy_value.md:80]: Is there another filter example we can use here that'll filter on different criteria than the 3-4 we're showing above?

File: docs/tables/turbot_policy_value.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+```
+
+### Extract all policy values from Turbot
```
Comment (cbruno10) [inline docs/tables/turbot_policy_value.md:83]: I don't believe we should include this example - policy values are set for every policy type on every applicable resource. I find it extremely rare that I'd want to see all policy values (policy settings or resources make more sense).

File: docs/tables/turbot_policy_value.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+of these columns: `id`, `resource_type_id`, `resource_type_uri` or `filter`.
+
+A Policy Value is the effective policy setting on an instance of a resource type. Every resource that is targeted by a given policy setting will have its own value for that policy, which will be the resultant calculated policy for the "winning" policy in the hierarchy.
+
+Policy settings are inherited through the resource hierarchy, and values for a resource are calculated according to policy settings at or above it in the resource hierarchy. For example, a policy setting at the Turbot level will be inherited by all resources below.
+
+It is recommended that queries to this table should include (usually in the `where` clause) at least one
+of these columns: `state`,`policyTypeId`, `resource_type_id`, `resource_type_uri` or `filter`.
+
+## Examples
```
Comment (cbruno10) [inline docs/tables/turbot_policy_value.md:16]: Can you please add examples using `is_default` and/or `is_calculated` (or any other columns that have interesting data about a policy value)? When coming up with these examples, please try to think as if you're a heavy Turbot user and want to pull out useful information from your workspace.

File: turbot/table_turbot_policy_value.go [hunk 12]
```diff
@@ -0,0 +1,211 @@
...
+		},
+		Columns: []*plugin.Column{
+			// Top columns
+			{Name: "id", Type: proto.ColumnType_INT, Transform: transform.FromField("Turbot.ID"), Description: "Unique identifier of the policy value."},
+			{Name: "policy_type_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Type.Title"), Description: "Title of the policy type."},
+			{Name: "poliy_type_trunk_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Type.Trunk.Title"), Description: "Title with full path of the policy type."},
+			{Name: "is_default", Type: proto.ColumnType_BOOL, Transform: transform.FromField("Default"), Description: "Defines the policy value is default or not."},
+			{Name: "is_calculated", Type: proto.ColumnType_BOOL, Description: "True if this is a policy setting will be calculated for each value."},
+			{Name: "precedence", Type: proto.ColumnType_STRING, Description: "Precedence of the setting: REQUIRED or RECOMMENDED."},
+			{Name: "resource_trunk_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Resource.Trunk.Title"), Description: "Full title (including ancestor trunk) of the resource."},
```
Comment (cbruno10) [inline turbot/table_turbot_policy_value.go:35]: Should this be below in the other `// Other columns` section, along with `resource_type_id`?

File: turbot/table_turbot_policy_value.go [hunk 13]
```diff
@@ -0,0 +1,211 @@
...
+				{Name: "filter", Require: plugin.Optional},
+			},
+		},
+		Columns: []*plugin.Column{
+			// Top columns
+			{Name: "id", Type: proto.ColumnType_INT, Transform: transform.FromField("Turbot.ID"), Description: "Unique identifier of the policy value."},
+			{Name: "policy_type_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Type.Title"), Description: "Title of the policy type."},
+			{Name: "poliy_type_trunk_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Type.Trunk.Title"), Description: "Title with full path of the policy type."},
+			{Name: "is_default", Type: proto.ColumnType_BOOL, Transform: transform.FromField("Default"), Description: "Defines the policy value is default or not."},
+			{Name: "is_calculated", Type: proto.ColumnType_BOOL, Description: "True if this is a policy setting will be calculated for each value."},
```
Comment (cbruno10) [inline turbot/table_turbot_policy_value.go:33]: Similar to `is_default`, and other columns in this table, the description does not have correct grammar, what description does our GraphQL API have?

File: turbot/table_turbot_policy_value.go [hunk 14]
```diff
@@ -0,0 +1,211 @@
...
+				{Name: "resource_type_id", Require: plugin.Optional},
+				{Name: "filter", Require: plugin.Optional},
+			},
+		},
+		Columns: []*plugin.Column{
+			// Top columns
+			{Name: "id", Type: proto.ColumnType_INT, Transform: transform.FromField("Turbot.ID"), Description: "Unique identifier of the policy value."},
+			{Name: "policy_type_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Type.Title"), Description: "Title of the policy type."},
+			{Name: "poliy_type_trunk_title", Type: proto.ColumnType_STRING, Transform: transform.FromField("Type.Trunk.Title"), Description: "Title with full path of the policy type."},
+			{Name: "is_default", Type: proto.ColumnType_BOOL, Transform: transform.FromField("Default"), Description: "Defines the policy value is default or not."},
```
Comment (cbruno10) [inline turbot/table_turbot_policy_value.go:32]: This description does not have correct grammar, what description does our GraphQL API have?

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@ParthaI Please see comments, thanks!

File: turbot/table_turbot_policy_value.go (hunk 12 above)
Comment (ParthaI) [reply turbot/table_turbot_policy_value.go:35]: It should be here, we should move `resource_type_id` column to here only

File: turbot/table_turbot_policy_value.go (hunk 13 above)
Comment (ParthaI) [reply turbot/table_turbot_policy_value.go:33]: We have description for is_calculated is `If true this value is derived from calculated setting inputs e.g. templateInput and template.` in GraphQL API.

Comment (cbruno10) [reply turbot/table_turbot_policy_value.go:33]: @ParthaI Let's copy the descriptions, they read OK in a description context here, and being able to re-use them is an easy way to get accurate, grammatically correct descriptions.

File: turbot/table_turbot_policy_value.go (hunk 12 above)
Comment (cbruno10) [reply turbot/table_turbot_policy_value.go:35]: Ok makes sense, let's move up `resource_id` column then

File: docs/tables/turbot_policy_value.md (hunk 9 above)
Comment (ParthaI) [reply docs/tables/turbot_policy_value.md:80]: Yes, we can use the filter `state:ok`.

File: docs/tables/turbot_policy_value.md [hunk 15]
```diff
@@ -0,0 +1,93 @@
+# Table: turbot_resource
+
+Resources in Turbot represent cloud configuration items such as users,
```
[the remark this answers is not in the document]
Comment (ParthaI) [reply docs/tables/turbot_policy_value.md:3]: Removed the description.

File: docs/tables/turbot_policy_value.md (hunk 3 above)
Comment (ParthaI) [reply docs/tables/turbot_policy_value.md:9]: Updated

File: docs/tables/turbot_policy_value.md (hunk 4 above)
Comment (ParthaI) [reply docs/tables/turbot_policy_value.md:18]: Updated

File: docs/tables/turbot_policy_value.md (hunk 5 above)
Comment (ParthaI) [reply docs/tables/turbot_policy_value.md:34]: Updated

File: docs/tables/turbot_policy_value.md (hunk 6 above)
Comment (ParthaI) [reply docs/tables/turbot_policy_value.md:31]: Added the `policy_type_id` value

File: docs/tables/turbot_policy_value.md (hunk 7 above)
Comment (ParthaI) [reply docs/tables/turbot_policy_value.md:47]: Updated

File: docs/tables/turbot_policy_value.md (hunk 8 above)
Comment (ParthaI) [reply docs/tables/turbot_policy_value.md:50]: Removed this example

File: docs/tables/turbot_policy_value.md (hunk 9 above)
Comment (ParthaI) [reply docs/tables/turbot_policy_value.md:80]: Except the filter keys which are mentioned in table doc other are throwing the error `Invalid filter key entered`

File: docs/tables/turbot_policy_value.md (hunk 10 above)
Comment (ParthaI) [reply docs/tables/turbot_policy_value.md:83]: Removed this example

File: turbot/table_turbot_policy_value.go (hunk 12 above)
Comment (ParthaI) [reply turbot/table_turbot_policy_value.go:36]: Moved the column

File: turbot/table_turbot_policy_value.go (hunk 13 above)
Comment (ParthaI) [reply turbot/table_turbot_policy_value.go]: Updated the description for `is_default` and `is_calculated` columns

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**